pkg/util/containers/providers/cgroup/cgroup_metrics.go

```diff
@@ -384,11 +376,10 @@ func (c ContainerCgroup) IO() (*metrics.ContainerIOStats, error) {
 		devices = mapping.idToName
 	}
 
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		fields := strings.Split(scanner.Text(), " ")
+	err = c.scanStatFile("blkio", "blkio.throttle.io_service_bytes", func(line string) error {
+		fields := strings.Split(line, " ")
 		if len(fields) < 3 {
-			continue
+			return nil
```

Comment:
Trying to understand under which circumstances we would continue here.
Looking at this cgroup it's clearly always 3 elements until the total
```
7:0 Async 0
7:0 Total 0
Total 29495738368
```

Reply:
Yes, it's to prevent any issue in case of malformed/unsupported syntax